### ISSUE TITLE
Not understand plus_in = inv_stdv * (centered_x + 1./255.) in discretized_mix_logistic_loss

### ISSUE THREAD

**Issue by blues-lin** [openai/pixel-cnn#33]
Could anybody explain why here is + 1./255. not 0.5 as paper equation (2) ?
```
60.    centered_x = x - means
61.    inv_stdv = tf.exp(-log_scales)
62.    plus_in = inv_stdv * (centered_x + 1./255.)
63.    cdf_plus = tf.nn.sigmoid(plus_in)
64.    min_in = inv_stdv * (centered_x - 1./255.)
65.    cdf_min = tf.nn.sigmoid(min_in)
```

**Comment by powder21** [comment]
input to pixelCNN is scaled from uint8 [0,255] to float in range [-1,1]
so the original value divide by 255/2, and 0.5 should be 0.5/(255/2) = 1/255

**Comment by blues-lin** [comment]
Here I'm still confused, for the model, inputs are values between [-1, 1]. And function sample_from_discretized_mix_logistic output in [-1, 1] which means model output the same range as inputs. Why should model need knowledge about original data scale (unscaled data)?

**Comment by powder21** [comment]
plotting.img_stretch scales its input to [0,1] as output

**Comment by sharvil** [comment]
I agree with @blues-lin; the expression should be +/-0.5 / 255. Shifting and changing the scale of the input doesn't change the number of discretized buckets. The probability of x should be computed with a bucket width of 1 centered around x, i.e. x +/-0.5 * bucket_width.

**Comment by powder21** [comment]
> I agree with @blues-lin; the expression should be +/-0.5 / 255. Shifting and changing the scale of the input doesn't change the number of discretized buckets. The probability of x should be computed with a bucket width of 1 centered around x, i.e. x +/-0.5 / (buckets-1).

width of +/-0.5 / 255 is 1/255, and there will be 2/(1/256)=512 buckets, which is 256 in paper.
